lib/linked_list.rb

```diff
@@ -18,71 +18,148 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(value)
-      raise NotImplementedError
+        new_node = Node.new(value)
+      if @head.nil?
+        @head = new_node
+      else 
+        new_node.next = @head
+        @head = new_node
+      end

```

Comment:
This can be simplified as:

```suggestion
       @head = Node.new(value, @head)
```